Re-cast as a Q&A (#3)
researchapps/betterscientificsoftware.github.io: Articles/Blog/2023-04-developer-surveys.md

You really can't put people in a check box, nor can you ask them to express a full need or sentiment in one.

*Q*: Did you test and revise the survey before launching it?

*Vanessa*: Yes. Once I had created all of my questions, I put them into a Microsoft Form (what we have to use) and then had the RADIUSS team test and give feedback. As an aside, if you are able to use Google Forms I highly recommend this over what we had to use, because I found Microsoft Forms much harder to use. We had a few rounds of feedback, and also sent it out for some testing by a set of volunteers. As you might expect, many questions were refined, simplified, or entirely removed. Our perspective at this time was to do our best, and not worry if it wasn't perfect.
We would learn from the experience and try differently the next time. A few months before the end of 2022, the survey was ready.

*Q*: How did you launch the survey and gather results?

*Vanessa*: We sent out the survey in the last few months of 2022, making sure to give it fun branding, and advertise it on many lists.
We sent out a few reminders, and gave people about 2 months to complete it.

> While we don't have a definite count of our reach because the survey was sent out on many platforms, I suspect the primary demographic we hit was Livermore Computing and our Town Hall channel.
> I would guesstimate a few hundred people.

Reviewer: I think we should just remove this - it was my mistake to guess the demographic we hit (we actually have a breakdown, it wasn't shared publicly) I think it sounds OK just removed!

```suggestion
```

At the end of the day, we had over 100 responses, half of which were self-reported developers.
About a quarter were staff scientists (that likely worked with code) and the remainder were group leaders, managers, technical support, or other support staff.
We had more participation than the application surveys of previous years.
So, we were relatively happy with the response.

*Q*: How did you analyze and present the results?

*Vanessa*: I wanted to host the results on a Jekyll web site.

_**Editor's note**: We would link to the site from this page but it is available on LLNL's internal network only. See below for a link to some of the publicly available results._

> *Vanessa*: Over the 2022 Christmas break, I started the web interface and results data parsing logic...even before the survey had closed.

Reviewer: ```suggestion
*Vanessa*: Over the 2022 Christmas break, I started the web interface and results data parsing logic... even before the survey had closed.
```

What you can't see from any web interface is that the plots are programmatically generated from the raw data, and the raw data is parsed from an exported comma separated value (csv) file.

> Plots are generated via Jeklyll [include snippets](https://jekyllrb.com/docs/includes/) so the logic for each plot type only needs to be written once.

Reviewer: ```suggestion
Plots are generated via Jekyll [include snippets](https://jekyllrb.com/docs/includes/) so the logic for each plot type only needs to be written once.
```

This is where we shared fun things we learned like favorite languages, editors, etc.

*Q*: What were some of the key take-aways or surprises from the survey?

*Vanessa*: You can see the [highlights](https://software.llnl.gov/radiuss/2023/02/08/first-developer-survey/) of our survey on the public facing page, and notably these tend to be positive oriented things.
We chose to not share challenges or gripes, which would (maybe ironically) be the most valuable information for us.
The feedback that came in is what you would expect - an overwhelmingly positive flow of gratitude for people and quality of work, and gripes about developer environments and/or tooling.
I say "as you would guess" because this is the classic problem dichotomy at an academic institution or a national lab.
Anywhere where shared high performance computing environments are present, there is a dual need to be conservative and consistent, and that is challenged by the constant reality and drive for change and new things.

*Vanessa*: One of the biggest challenges in this process isn't necessarily knowing what should be worked on, but figuring out how to do that.

> As an example, the RADIUSS team is a small group of people, each only 10-50% of time devoted to RADIUSS, and often in the context of a specific code project used in RADIUSS.

Reviewer: ```suggestion
As an example, the RADIUSS team is a small group of people, each only 10-50% of time devoted to RADIUSS, and often in the context of a specific code project under the RADIUSS umbrella.
```

*Q:* Is there anything you would do differently?

*Vanessa*: One of my greatest worries about the survey is that we might put it out, get great insights, but not be empowered to take action on many of them.
I'm hopeful that if we identify many problems, if we can minimally target two to three of them a year to work on for that year, we can slowly make progress toward better developer experience and happiness.

*Q*: Any closing thoughts?

*Vanessa*: You can read our first public set of highlights at [the RADIUSS news](https://software.llnl.gov/radiuss/2023/02/08/first-developer-survey/)
blog.
We hope that other academic institutions and/or national labs take interest in the happiness and productivity of their developers, and consider contributing to this space.
An important point to remember is that while we may not be able to respond to every desire or gripe, the fact that we are asking, period, means that we are trying our best, and our hearts are in the right place.
Even large, meaningful changes can start with small steps, and this kind of effort is exactly that.

> Perhaps if we could identify shared gripes, the combined forces of multiple institutions would have enough people power to work on them?

Reviewer: This sounds better as an ending (instead of ending on the question).
```suggestion
Perhaps if we could identify shared gripes, the combined forces of multiple institutions would have enough people power to work on them?
This is something I would be excited about.
```
